android/src/main/java/com/facebook/reactnative/androidsdk/FBAppEventsLoggerModule.java

```diff
@@ -216,6 +219,40 @@ public String getUserID() {
        return mAppEventLogger.getUserID();
      }
 
+     /**
+      * Each app/device pair gets an GUID that is sent back with App Events and persisted with this
+      * app/device pair.
+      *
+      * @return The GUID for this app/device pair.
+      */
+     @ReactMethod(isBlockingSynchronousMethod = true)
```

Comment:
I think we can make all the functions not blocking and use the promise API since they are async in JS anyway.